src/applications/gi/components/ResponsiveTable.jsx

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+
+const borderClasses =
+  'vads-u-border-top--0 vads-u-border-right--0 vads-u-border-left--0 vads-u-padding--0 vads-u-padding-y--0p5 medium-screen:vads-u-padding--1';
+const rowPaddingClass = 'vads-u-padding-y--2';
+
+class ResponsiveTable extends React.Component {
+  renderHeader = field => {
+    return (
+      <th key={field} className={borderClasses} role="columnheader">
+        {field}
+      </th>
+    );
+  };
+
+  renderRow = item => {
+    const { fields } = this.props;
+    let extraClass = '';
+    return (
+      <tr
+        key={item.id}
+        className={`${borderClasses} ${rowPaddingClass}`}
+        role="row"
+      >
+        {fields.map((field, index) => {
+          // This is to right align the amount field and account number fields
+          // since they are numeric
+          if (index === 1 || index === 5) {
+            extraClass =
+              'vads-u-text-align--left medium-screen:vads-u-text-align--right';
+          } else {
+            extraClass = '';
+          }
+          return (
+            <td
+              data-index={index}
+              className={`${borderClasses} ${extraClass}`}
+              data-label={`${field}:`}
+              key={`${item.id}-${field}`}
+              role="cell"
+            >
+              {item[field]}
+            </td>
+          );
+        })}
+      </tr>
+    );
+  };
+
+  render() {
+    const { fields, data, tableClass } = this.props;
+    const headers = fields.map(this.renderHeader);
+    const rows = data?.map(this.renderRow());
+    const classes = classNames('responsive', tableClass);
+
+    return (
+      <table className={classes} role="table">
+        {/* eslint-disable-next-line jsx-a11y/no-redundant-roles */}
+        <thead role="rowgroup">
+          <tr role="row">{headers}</tr>
+        </thead>
+        {/* eslint-disable-next-line jsx-a11y/no-redundant-roles */}
+        <tbody role="rowgroup">
+          {this.props.children}
+          {rows}
+        </tbody>
+      </table>
+    );
+  }
+}
+
+ResponsiveTable.propTypes = {
+  fields: PropTypes.arrayOf(PropTypes.string),
+  data: PropTypes.arrayOf(
+    PropTypes.shape({
+      id: PropTypes.string,
+    }),
+  ),
+  tableClass: PropTypes.string,
+};
+
+export default ResponsiveTable;
```

Comment:
Is this the same component as `src/applications/disability-benefits/view-payments/components/responsive-table/ResponsiveTable.jsx`? We're just not sharing the code because they're in app directories?

Reply:
The components are slightly different. I did not create a component in platform because of https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/203#issuecomment-679138608

Reply:
Ah, bummer that we couldn't share it

Reply:
A component in platform could be created but it seemed creating a robust one (neither version is robust enough for platform at the moment) was outside of the scope for what was needed as a solution for https://github.com/department-of-veterans-affairs/va.gov-team/issues/11055